### PR TITLE
Invite command & code cleaning and improvement

### PR DESCRIPTION
**Pull request by ysengoku**
### Usage
👉🏻  `INVITE clientNick #channel`   
Invite a client to the #channel.

👉🏻  `INVITE` (without parameter / Only on netcat)
Receive the list of channels that the client was invited to.

- This command can be executed only by channel operators. 
- When `INVITE` is accepted, the invited client is registered in `clientPMap _invitedClients` of the channel. He is not directly added to the channel until he sends `JOIN`.
- If the invited-only mode is activated, only invited clients can participate in the channel after `JOIN`. (Need to test after `MODE` command implementation)

---

### Client modes
Everything related to Client modes is removed (commented for the moment) because we don't handle Client modes.

### Secured `send()` function
The last parameter of send() function is set to `MSG_NOSIGNAL` instead of 0 (default) thanks to advice from Camille.

Explanation:
`MSG_NOSIGNAL flag` prevents send from generating a `SIGPIPE signal` if the peer has closed the connection.
Instead of terminating the process, send will simply return -1, and errno will be set to EPIPE. This allows the program to handle the error gracefully without dealing with a potentially unhandled signal that could stop the program.

### CODE REVIEW

File: src/commands/invite.cpp [hunk 1]
```diff
@@ -6,96 +6,86 @@
...
-    } else {
-      channel.addClientToInvitedMap(invitedClient);
-      send341Inviting(_clients.at(invitingClientFd), nick, channel);
-      invitedClient->receiveMessage(":" + \
-      _clients.at(invitingClientFd).getNickname() + " INVITE " + nick + " :" \
-      + channel.getNameWithPrefix() + "\r\n");
-    }
+    channel.addClientToInvitedMap(invitedClient);
+    send341Inviting(_clients.at(invitingClientFd), invitedNick, channel);
+    invitedClient->receiveMessage(
```
Comment (faboussard) [inline src/commands/invite.cpp:76]: si tu veux tu peux utiliser ma fonction "sendToClient" du fichier privmsg.cpp

**Review by faboussard — APPROVED** [review]

File: src/commands/invite.cpp (hunk 1 above)
Comment (ysengoku) [reply src/commands/invite.cpp:76]: Au depart, on s'est dit que l'on utiliserait cette function `receiveMessage()` dans la class Client pour simplifier les choses pour tous les messages envoyes aux clients. Et elle est deja utilise partout...

@faboussard @melobern  Il faudra que l'on choisisse soit `receiveMessage()` soit  `sendToClient`.

File: includes/Client.hpp [hunk 2]
```diff
@@ -43,7 +43,8 @@ class Client {
   bool _realnameSet;
   bool _passwordGiven;
   bool _accepted;
-  UserModes _uModes;
+  bool _CapSend;
```
Comment (melobern) [inline includes/Client.hpp:46]: I suppressed this on my last pull request. Did you merge the main properly ? Otherwise we should delete this.

Comment (ysengoku) [reply includes/Client.hpp]: Removed this line !

**Pull request merged**